sites/public/lib/helpers.tsx

```diff
@@ -60,6 +62,21 @@ export const getImageTagLabelFromListing = (listing: Listing) => {
     : undefined
 }
 
+export const getListingTags = (listing: Listing) => {
+  const tags: ImageTag[] =
+    listing?.listingPrograms?.map((program) => {
+      return { text: program.program.title }
+    }) ?? []
+  if (accessibilityFeaturesExist(listing)) {
+    tags.push({
+      text: t("listings.reservedCommunityTypes.specialNeeds"),
+      iconType: "universalAccess",
+      iconColor: AppearanceStyleType.primary,
+    })
+  }
+  return tags
+}
```

Comment:
Along with splitting out the section added to ListingView, if this was added to that file with a useMemo hook that had listing?.listingPrograms and listing?.features as dependencies, we could reduce the amount of times this piece needs to render. So this and `accessibilityFeaturesExist` would need to be updated accordingly to not accept the full listing.

Reply:
That is a very valid point, I think the only dynamic thing going on the page is checking if a user has a profile, so at most it would probably render twice. With that, I agree that we can hold off on using useMemo until/if there is a clear performance gain.